marklogic-client-api/src/main/java/com/marklogic/client/datamovement/JSONSplitter.java

```diff
@@ -128,11 +161,20 @@ public long getCount() {
      * @param input JsonParser created from the JSON file
      * @return a stream of DocumentWriteOperation to write to database
      */
-    public Stream<DocumentWriteOperation> splitWriteOperations(JsonParser input) {
+    public Stream<DocumentWriteOperation> splitWriteOperations(JsonParser input, String inputName) {
         if (input == null) {
             throw new IllegalArgumentException("Input cannot be null");
         }
 
+        if (getUriMaker() == null) {
+            JSONSplitter.UriMakeImpl uriMaker = new UriMakeImpl();
+            setUriMaker(uriMaker);
+        }
+        getUriMaker().setInputName(inputName);
+        if (getUriMaker().getInputName() == null) {
+            ((JSONSplitter.UriMakeImpl) getUriMaker()).setExtension("json");
```

Comment:
I think we should keep this part here. Here's why.

I add this section because we have both splitWriteOperations(InputStream input) and splitWriteOperations(InputStream input, String inputName). Then the user only calls the first API and it will work. Or you mean if they call the first API, they should provide their own URI maker to make the whole thing work. Is this what you mean?

The purpose of line 174 through 176 is to set the correct extension to the generated URI when the user calls splitWriteOperations(InputStream input) without an input name. This only applies to JSONSplitter and XMLSplitter. So I don't think it should be put into UriMakeImpl.setInputName().



That's why I think we should keep this part here.

Reply:
Discussed offline, moved line 174-176 to JSONSplitter.UriMakeImpl class.